dev_tools/tests/test_nxdl_utils.py

```diff
@@ -194,3 +194,41 @@ def test_namefitting_precedence(better_fit, better_ref, worse_fit, worse_ref):
     assert nexus.get_nx_namefit(better_fit, better_ref) > nexus.get_nx_namefit(
         worse_fit, worse_ref
     )
+
```

Comment:
didn't we wanted to put the test to pynxtools?

Reply:
The problem is: I cannot just copy all the functionality to pynxtools and just call the function here, then I'll get an import error. But since we don't want numpy here, I ended up splitting the functionality. Here, it checks for bytes and list of bytes and in pynxtools, it calls this function from the nxdl_utils, but on top also performs conversion on np arrays.

Therefore, the tests here do the checking for the bytes and list decoding.